packages/dev/core/src/Materials/meshDebugPluginMaterial.ts

```diff
@@ -0,0 +1,442 @@
+                        return min(min(a3.x, a3.y), a3.z);
+                    }
+                #endif
+
+                #if MODE == 1
+                    float m_cornerFactor() {
+                        vec3 worldPos = vPositionW;
+                        float dist = length(worldPos - vVertexWorldPos);
+                        float camDist = length(worldPos - vEyePosition.xyz);
+                        float d = sqrt(camDist) * 0.001;
+                        return smoothstep((m_vertexRadius * d), ((m_vertexRadius * 1.01) * d), dist);
+                    }
+                #endif
+
+                #if (MODE == 2 && defined(UV1)) || (MODE == 3 && defined(UV2))
+                    float m_checkerboardFactor(vec2 uv) {
+                        vec2 f = fract(uv * m_uvScale);
+                        f -= .5;
+                        return (f.x * f.y) > 0. ? 1. : 0.;
+                    }
+                #endif
+            `,
+                CUSTOM_FRAGMENT_MAIN_END: `
+                vec3 m_color = m_shadedDiffuseColor;
+
+                #if MODE == 0
+                    m_color = mix(m_wireframeColor1, m_color, m_edgeFactor());
+                #elif MODE == 1
+                    float cornerFactor = m_cornerFactor();
+                    if (vPass == 0. && cornerFactor == 1.) discard;
+                    m_color = mix(m_wireframeColor2, m_color, m_edgeFactor());
+                    m_color *= mix(m_wireframeColor1, m_color, m_cornerFactor);
+                #elif MODE == 2 && defined(UV1)
+                    m_color = mix(m_uvColor1, m_uvColor2, m_checkerboardFactor(vMainUV1));
+                #elif MODE == 3 && defined(UV2)
+                    m_color = mix(m_uvColor1, m_uvColor2, m_checkerboardFactor(vMainUV2));
+                #elif MODE == 4 && defined(VERTEXCOLOR)
+                    m_color = vColor.rgb;
+                #elif MODE == 5
+                    m_color = m_materialColor;
+                #endif
+
+                #if DEBUGMODE == 0
+                    #if MODE != 4
+                        gl_FragColor = vec4(m_applyShading(m_color), 1.);
+                    #else
+                        gl_FragColor = vec4(m_color, 1.);
+                    #endif
+                #elif defined(MULTIPLYDEBUG)
+                    gl_FragColor *= vec4(m_color, 1.);
+                #endif
+            `,
+              };
+    }
+
+    /**
+     * Generate a random color
+     * @returns Random RGB color as a Vector3
+     */
+    private _getRandomColor(): Vector3 {
+        const hue: number = Math.random() * 360;
+        const color: Color3 = Color3.FromHSV(hue, 0.95, 0.99);
+        return Vector3.FromArray(color.asArray());
+    }
+
+    /**
+     * Renders triangles in a mesh 3 times by tripling the indices in the index buffer.
+     * Used to prepare a mesh to be rendered in TRIANGLES or VERTICES modes.
+     * NOTE: This is a destructive operation. The mesh's index buffer will be modified.

```

Comment:
You should also add that the vertex buffers are also modified and that a new vertex buffer is also created.